internal/pkg/manifest/svc.go

```diff
@@ -73,18 +73,68 @@ func (a *Count) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		}
 	}
 
-	if !a.Autoscaling.IsEmpty() {
+	if !c.Autoscaling.IsValid() {
+		return errUnmarshalSpot
+	}
+
+	if !c.Autoscaling.IsEmpty() {
+		// Successfully unmarshalled Autoscaling fields, return
 		return nil
 	}
 
-	if err := unmarshal(&a.Value); err != nil {
+	if err := unmarshal(&c.Value); err != nil {
 		return errUnmarshalCountOpts
 	}
 	return nil
 }
 
+// Spot is a custom type that can either be represented as a boolean, when
+// specified with a Range, or as an integer, which indicates the `desiredCount`
+// on a service that does not use application autoscaling.
+type Spot struct {
+	Enabled *bool
+	Base    *int
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the Spot
+// struct, allowing it to be unmarshalled into either an int value provided by
+// Base or a boolean value provided by Enabled.
+// This method implements the yaml.Unmarshaler (v2) interface.
+func (s *Spot) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if err := unmarshal(&s.Base); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if *s.Base != 0 {
+		s.Enabled = nil
+		return nil
+	}
+
+	if err := unmarshal(&s.Enabled); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if s.Enabled != nil {
```

Comment:
(If we need to use `if *s.Base != 0` as opposed to `if s.Base != nil` above, then please ignore this comment)

Would `s.Base`  have been `nil` anyway if we get to this point?

Reply:
Without this block, I get this error:
```
$ go test ./internal/pkg/manifest/ -run TestCount_UnmarshalYAML
--- FAIL: TestCount_UnmarshalYAML (0.00s)
    --- FAIL: TestCount_UnmarshalYAML/With_spot_enabled_with_autoscaling_range (0.00s)
        svc_test.go:344: 
            	Error Trace:	svc_test.go:344
            	Error:      	Received unexpected error:
            	            	cannot unmarshal "spot" field to an integer or boolean
            	Test:       	TestCount_UnmarshalYAML/With_spot_enabled_with_autoscaling_range
FAIL
```